### PR TITLE
feat: adds tokensProxy when swapping with dex

### PR DESCRIPTION
**Pull request by 0xged**

### CODE REVIEW

**Comment by 0xged** [comment]
@nchamo just check changes on DCAHubCompanion. Most of changed files come from PR #67 that fixed integrations tests.

File: contracts/interfaces/IDCAHubCompanion.sol [hunk 1]
```diff
@@ -105,6 +105,7 @@ interface IDCAHubCompanionSwapHandler is IDCAHubSwapCallee {
 
   /// @notice Executes a swap with the given DEX, and sends all unspent tokens to the given recipient
   /// @param _dex The DEX that will be used in the swap
+  /// @param _tokensProxy The spender of the tokens (we will not always need to approve the dex)
```
Comment (nchamo) [inline contracts/interfaces/IDCAHubCompanion.sol:108]: I think this is a little more clear
```suggestion
  /// @param _tokensProxy The spender of the tokens (could be different from the dex)
```

File: contracts/interfaces/IDCAHubCompanion.sol [hunk 2]
```diff
@@ -126,6 +128,7 @@ interface IDCAHubCompanionSwapHandler is IDCAHubSwapCallee {
   /// @notice Executes a swap with the given DEX and sends all `reward` unspent tokens to the given recipient.
   /// All positive slippage for tokens that need to be returned to the hub is also sent to the hub
   /// @param _dex The DEX that will be used in the swap
+  /// @param _tokensProxy The spender of the tokens (we will not always need to approve the dex)
```
Comment (nchamo) [inline contracts/interfaces/IDCAHubCompanion.sol:131]: I think this is a little more clear
```suggestion
  /// @param _tokensProxy The spender of the tokens (could be different from the dex)
```

**Review by nchamo — APPROVED** [review]
Looks good!

**Pull request merged**